pkg/app/pprof.go

```diff
@@ -0,0 +1,52 @@
+package app
+
+import (
+	"fmt"
+	"net/http"
+	_ "net/http/pprof" //nolint:gosec // Import for pprof, only enabled via CLI flag
+	"strings"
+	"time"
+)
+
+type Server interface {
+	Start(string)
+	ErrChan() <-chan error
+}
+
+type PprofServer struct {
+	err chan error
+}
+
+func NewPprofServer() *PprofServer {
+	return &PprofServer{
+		err: make(chan error),
+	}
+}
+
+func (p *PprofServer) Start(address string) {
+	go func() {
+		pprofHostPort := address
+		parts := strings.Split(pprofHostPort, ":")
+		if len(parts) == 2 && parts[0] == "" {
+			pprofHostPort = "localhost:" + parts[1]
+		}
+		pprofHostPort = "http://" + pprofHostPort + "/debug/pprof"
+
+		fmt.Printf("I! Starting pprof HTTP server at: %s\n", pprofHostPort)
+
+		server := &http.Server{
+			Addr:         address,
+			ReadTimeout:  10 * time.Second,
```

Comment:
10 seconds for WriteTimeout is short, leave it unset or set it to 1 minute or higher.
Pprof does not expect a request body, set `ReadHeaderTimeout` to 5 or 10 seconds and leave ReadTimeout unset.